src/devtools/client/debugger/src/utils/sourceVisualizations.ts

```diff
@@ -92,23 +77,28 @@ export function getUniqueAlternateSourceId(sourceId: string): {
   why?: "no-sourcemap" | "not-unique";
 } {
   const generatedSourceIds = ThreadFront.getGeneratedSourceIds(sourceId);
-  if (!generatedSourceIds?.length) {
-    return { why: "no-sourcemap" };
-  }
+  const nonPrettyPrintedSourceId =
```

Comment:
Only somewhat related: is it possible for both generated and original sources to have pretty-printed versions? If so, it seems like things get confusing?

Reply:
Yes, that's possible.

Reply:
That seems like something that the idea of "only one unique alternative source" does not seem to play nicely with. Does that seem correct to you?

Reply:
No, the logic for determining the "unique alternative source" basically conflates minified sources with their pretty-printed versions and then looks at "the other" source (because when you ignore pretty-printing, you're left with 2 sources again) and returns its pretty-printed version.

Reply:
I don't understand why it shouldn't work in that case. Maybe we should VC on that later.

Reply:
Yeah, I agree that there's nothing wrong with this implementation. I think (correct me if I'm wrong) that if we have 4 sources:

![Page 1 (3)](https://user-images.githubusercontent.com/5903784/176866188-a0918715-d80d-4873-8083-d0e47bcf0c63.png)

And we view `o1` and we ask for it's alternate we will get `pp1`. Is that right? And then if we ask for the alternate of that we will get `ppo1`. It's not a bug in your implementation, I think it's just a limitation of the idea of "there can only be one alternative" when in fact a source was actually the middle step in a series of transformations.

Reply:
> And we view o1 and we ask for it's alternate we will get pp1. Is that right? And then if we ask for the alternate of that we will get ppo1.

Yes, that is exactly what would happen. BUT in this scenario you should never be able to see `o1`, only `ppo1`. If you ever manage to open `o1`, that would be a bug!